### PR TITLE
Feature usecase 1 2 3 4 12

### PR DESCRIPTION
**Pull request by MashB**
Feature usecase 1 ,2, 3, 4, 12 changes updated

### CODE REVIEW

File: tdei-api-gateway.json [hunk 1]
```diff
@@ -181,30 +239,60 @@
                         "AuthorizationToken": []
                     }
                 ]
-            },
+            }
+        },
+        "/api/v1/osw/upload/{tdei_project_group_id}": {
```
Comment (sureshgaussian) [inline tdei-api-gateway.json:244]: how is the mobility service ID taken? Use case says: For WSDOT project_group, KCM Mobility service, OSW data is being uploaded. It seems like you assumed only project_group is necessary for OSW. Change.

File: tdei-api-gateway.json [hunk 2]
```diff
@@ -181,30 +239,60 @@
...
             "post": {
                 "tags": [
                     "OSW"
                 ],
-                "summary": "Path used to upload/create a new osw data file.",
+                "summary": "upload/create a new osw data file.",
```
Comment (sureshgaussian) [inline tdei-api-gateway.json:249]: upload a pre-release of OSW dataset

File: tdei-api-gateway.json [hunk 3]
```diff
@@ -243,121 +335,59 @@
...
+            "post": {
                 "tags": [
-                    "GTFS-Pathways"
+                    "OSW"
                 ],
-                "summary": "List pathways files meeting criteria.",
-                "description": "This endpoint returns a json list of all gtfs pathways files stored in the TDEI system that meet the specified criteria. Criteria that can be specified include: polygon (bounding box), minimum confidence level, pathways version, date time and agency id.\n\nThis endpoint can be used by an application developer to obtain a list of gtfs pathways files in the TDEI system meeting the specified criteria. This endpoint returns a list of file-metadata including the uris of the file, which can be used to fetch the files themselves.",
-                "operationId": "listPathwaysFiles",
+                "summary": "Publishes the OSW dataset for the tdei_record_id",
+                "description": "Publishes an OSW dataset that has undergone prior uploading and editing via the [POST] /osw endpoint, marking it as an official release for the uploaded mobility service. This official release status ensures visibility to all TDEI data consumers.",
```
Comment (sureshgaussian) [inline tdei-api-gateway.json:344]: Publishes an OSW dataset that was previously uploaded via the [POST] /osw endpoint, marking it as an official release for the mobility service. This official release status ensures visibility to all TDEI data consumers.

File: tdei-api-gateway.json [hunk 4]
```diff
@@ -370,28 +400,26 @@
...
             "post": {
                 "tags": [
-                    "GTFS-Pathways"
+                    "OSW"
                 ],
-                "summary": "create pathways file",
-                "description": "This call allows a user to upload or create a new gtfs pathways file. The caller must provide metadata about the file. Required metadata includes information about how and when the data was collected and valid dates of the file. Returns the tdei_record_id of the uploaded file.",
-                "operationId": "uploadPathwaysFile",
+                "summary": "Validates the osw dataset.",
+                "description": "Allows a user to validate osw dataset to check the correctness of data. The caller must provide metadata about the file - includes information about how and when the data was collected and valid dates of the file. Returns the job_id associated with validation job.",
```
Comment (sureshgaussian) [inline tdei-api-gateway.json:411]: we dont need the metadata for validating the dataset.

File: tdei-api-gateway.json [hunk 5]
```diff
@@ -432,253 +468,208 @@
                 ]
             }
         },
-        "/api/v1/gtfs-flex": {
-            "get": {
+        "/api/v1/osw/format/upload": {
```
Comment (sureshgaussian) [inline tdei-api-gateway.json:471]: change the name to 'osw/convert'

File: tdei-api-gateway.json [hunk 6]
```diff
@@ -432,253 +468,208 @@
...
-                        "description": "A bounding box which specifies the area to be searched. A bounding box is specified by a string providing the lat/lon coordinates of the corners of the bounding box. Coordinate should be specified as west, south, east, north.",
-                        "required": false,
-                        "schema": {
-                            "maxItems": 4,
-                            "minItems": 4,
-                            "type": "array",
-                            "items": {
-                                "type": "number",
-                                "format": "double"
+                "summary": "Request osw format on demand",
```
Comment (sureshgaussian) [inline tdei-api-gateway.json:476]: OSW reformatting on demand

File: tdei-api-gateway.json [hunk 7]
```diff
@@ -432,253 +468,208 @@
...
+                "description": "Upload a file and request for formatting on demand",
```
Comment (sureshgaussian) [inline tdei-api-gateway.json:477]: upload a file and request for file format conversion

File: tdei-api-gateway.json [hunk 8]
```diff
@@ -432,253 +468,208 @@
...
+                            "schema": {
+                                "required": [
+                                    "file",
+                                    "source",
+                                    "target"
+                                ],
+                                "type": "object",
+                                "properties": {
+                                    "source": {
+                                        "type": "string",
```
Comment (sureshgaussian) [inline tdei-api-gateway.json:491]: can't be string. I think this needs to be enum. we only have two options, right?

File: tdei-api-gateway.json [hunk 9]
```diff
@@ -432,253 +468,208 @@
...
+                                "$ref": "#/components/schemas/OSWConfidenceRequest"
                             }
                         }
                     },
                     "required": true
                 },
                 "responses": {
-                    "202": {
-                        "description": "The request has been accepted for processing.  returns the tdei_record_id, unique identifier for uploaded file.",
+                    "200": {
```
Comment (sureshgaussian) [inline tdei-api-gateway.json:552]: why is this 200, and not 202 like other async calls?

File: tdei-api-gateway.json [hunk 10]
```diff
@@ -687,21 +678,21 @@
                 ],
                 "responses": {
                     "200": {
-                        "description": "Returns status of record",
+                        "description": " Status of the formatting Job ",
                         "content": {
                             "application/json": {
                                 "schema": {
-                                    "$ref": "#/components/schemas/RecordStatus"
+                                    "$ref": "#/components/schemas/OSWFormatStatusResponse"
```
Comment (sureshgaussian) [inline tdei-api-gateway.json:685]: how does one download the reformatted output?

File: tdei-api-gateway.json [hunk 11]
```diff
@@ -729,7 +720,7 @@
                     {
                         "name": "format",
                         "in": "query",
-                        "description": "File format to download.",
+                        "description": "File format to download. Default to osw",
                         "required": false,
                         "schema": {
                             "type": "string",
```
Comment (sureshgaussian) [inline tdei-api-gateway.json:726]: enum?

**Review by sureshgaussian — CHANGES_REQUESTED** [review]
partial review of the file done.

File: tdei-api-gateway.json (hunk 1 above)
Comment (MashB) [reply tdei-api-gateway.json:244]: @sureshgaussian , Currently we do not have 'service_id' for OSW resource. If anything missing from requirement let me know I will include the change.

File: tdei-api-gateway.json (hunk 8 above)
Comment (MashB) [reply tdei-api-gateway.json:491]: @susrisha , fyi

File: tdei-api-gateway.json (hunk 9 above)
Comment (MashB) [reply tdei-api-gateway.json:552]: @susrisha , fyi

File: tdei-api-gateway.json (hunk 10 above)
Comment (MashB) [reply tdei-api-gateway.json:685]: @susrisha , fyi

File: tdei-api-gateway.json (hunk 5 above)
Comment (MashB) [reply tdei-api-gateway.json:471]: @susrisha , fyi

File: tdei-api-gateway.json (hunk 11 above)
Comment (MashB) [reply tdei-api-gateway.json:726]: Yes enum options, osm & osw

File: tdei-api-gateway.json (hunk 7 above)
Comment (MashB) [reply tdei-api-gateway.json]: Done

File: tdei-api-gateway.json (hunk 6 above)
Comment (MashB) [reply tdei-api-gateway.json]: Done

File: tdei-api-gateway.json (hunk 4 above)
Comment (MashB) [reply tdei-api-gateway.json]: Done

File: tdei-api-gateway.json (hunk 3 above)
Comment (MashB) [reply tdei-api-gateway.json]: Done

File: tdei-api-gateway.json (hunk 2 above)
Comment (MashB) [reply tdei-api-gateway.json]: Done

File: tdei-api-gateway.json [hunk 12]
```diff
@@ -63,35 +63,93 @@
...
+            }
+        },
         "/api/v1/osw": {
             "get": {
                 "tags": [
                     "OSW"
                 ],
                 "summary": "List osw files meeting criteria.",
-                "description": "This endpoint returns a list of url to gzip'd geojson files with osw data that meet the specified criteria. Criteria that can be specified include: a polygon (bounding box), minimum confidence level and osw version. This endpoint can be used by an application developer to obtain a list of osw files in the TDEI system meeting the specified criteria.",
+                "description": "This endpoint returns a list of url to gzip'd geojson files with osw data that meet the specified criteria. Criteria that can be specified include: a dataset area (polygon), minimum confidence level and osw version. This endpoint can be used by an application developer to obtain a list of osw files in the TDEI system meeting the specified criteria.",
```
[the remark this answers is not in the document]
Comment (MashB) [reply tdei-api-gateway.json]: Done

File: tdei-api-gateway.json (hunk 1 above)
Comment (sureshgaussian) [reply tdei-api-gateway.json:244]: @MashB , Use case 1 shows proj_group as WSDoT, mobility_service as KCM. So, we should add mobility_service for OSW.

File: tdei-api-gateway.json (hunk 8 above)
Comment (MashB) [reply tdei-api-gateway.json:512]: Done

File: tdei-api-gateway.json (hunk 10 above)
Comment (susrisha) [reply tdei-api-gateway.json:714]: Added ticket for the conversion job. Decided to give out another API for downloading the converted files. Will be implemented separately

**Review by susrisha — APPROVED** [review]
Approved

**Pull request merged**